psi4/9b60184.nix

```diff
@@ -0,0 +1,244 @@
+{ stdenv, buildPackages, makeWrapper, fetchFromGitHub, pkg-config, writeTextFile
+# Build time dependencies
+, cmake
+, perl
+, gfortran
+# Runtime dependencies
+, python
+, pybind11
+, qcelemental
+, qcengine
+, numpy
+, pylibefp
+, deepdiff
+, blas
+, lapack
+, gau2grid-2_0_4
+, libint1
+, libxc
+, dkh
+, dftd3
+, pcmsolver
+, libefp
+, chemps2
+, hdf5
+, hdf5-cpp
+# Optional
+, mrcc ? null
+}:
+let
+  useMKL = blas.passthru.implementation == "mkl";
```

Comment:
openblas vs. MKL: bagel also has the problem that some routines produce error when compiled against openblas (i.e. it only works properly with MKL). 
Here's what would suggest:
* Keep using the generic `lapack` and `blas` inputs but add an assert that fails in case it is not MKL.
* Call the package with something like this:
```
psi4 = callPackage { blas = blas.override { blasProvider = mkl; }; lapack = .... };
```

Reply:
I have added the assert for MKL. But as psi4 lives in the `pythonPackages.nix` and its `super` does not know BLAS, I cannot make this override there, as far as I can see. 

 The unwrap of the BLAS provider, that I am doing there with `blas_ = blas_ = blas.passthru.provider` and equivalently for LAPACK, is unfortunately required. Psi4's CMake misses some information about OpenMP libraries, when directly using `blas` and fails to link.

I view of both problems, it might be reasonable to directly pass MKL to Psi4? Alternatively i could use some hint how to do your suggested override in `pythonPackages.nix`.

Reply:
Oh I did not realise that is defined as python package. In that case: did you ever try to build it with `buildPythonPackage` instead of `mkDerivation`? That should take of most the python environment issues.

Reply:
To solve override we would need to fix the construction of how the override set is pulled in, which is probably a good idea. I ran into the some problem in another context.

This may get the job done(?):

`pythonPackages.nix`:
```
self: super: selfPkgs: superPkgs:
```
`default.nix`:
```
pythonOverrides = (import ./pythonPackages.nix) self super;
```

Reply:
> Oh I did not realise that is defined as python package. In that case: did you ever try to build it with `buildPythonPackage` instead of `mkDerivation`? That should take of most the python environment issues.

I gave it a quick try but was not too successfull. It actually is more a C++/Cmake project with python glue around it. There is no `setup.py` or anything similar, so i guess there would be lots of fiddling around with buildPythonPackage?

> This may get the job done(?):
Yes, this is a good idea and its working :slightly_smiling_face: